### DISCUSSION TITLE
Sandpack and shareable links

### DISCUSSION THREAD

**Discussion by NinoMaj** [codesandbox/codesandbox-client#4939] [General] [answered]
I have two questions about how Sandpack and shareable links works:
1. When I am building app with Sandpack package it is not available publicly until I call `getCodeSandboxURL` method? Could someone guess sandbox `id` and get to my code if I didn't call `getCodeSandboxURL`?
2. Is there a way to make app found on link I get from `getCodeSandboxURL` available publicly without exposing underlying code? Perhaps by having Codesandbox Pro account.

**Comment by siddharthkp** [comment]
@CompuIves This is right down your alley :)

**Comment by CompuIves** [comment, marked as answer]
Hey @NinoMaj!

Good questions. To answer:

1. No, everything will stay local to Sandpack. `getCodeSandboxURL` will generate the sandbox as soon as you call it, not before that. The ID won't be guessable.
2. It's possible to fork the sandbox generated by `getCodeSandboxURL` under a pro account and make it private, but the preview will only accessible by people who also have access to the code. The reason for this is the fact that we execute the code in the browser, so for the preview to actually show something it needs to download the code for execution. This means that savvy users will always be able to access the minified code.

I hope this answers your questions, let me know if you have any other questions!

**Reply by NinoMaj** [reply]
@CompuIves thanks for the answers.
1.) Ok, great to hear that :)
2.) What is the exact workflow? I generate sandbox by `getCodeSandboxURL` (initially it will be public), I take that url open it in `codesandbox`,  fork it and make it private. Does that mean that public version will still be available when users are searching all sandboxes? My ideally workflow would be to pass my codesandbox PRO account id or token to getCodeSandboxURL (`getCodeSandboxURL(id)`) and get link of private sandbox.

**Reply by CompuIves** [reply]
Hm, on point 2, we currently don't accept an API key for that sandbox. So the workflow you described would be the way to do it, but I can imagine that doesn't fit your use case since a public sandbox is created.

What is your use case? Maybe we can build something for it.

**Reply by NinoMaj** [reply]
We have a clients building websites in browsers with Sandpack package and we want to enable preview feature for pages they are building. Idea is that they should be able to just press a `generate preview` button,  get a link which will be generated by `getCodeSandboxURL` and they can send it to their clients.
